### PR TITLE
Adding Validation

### PR DESCRIPTION
**Pull request by divpaul2016**

### CODE REVIEW

File: src/main/stgeorge/validator/RequestValidate.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    public  List<RequestValidationResult> validateRequest()
+    {
+        List<RequestValidationResult> results = new ArrayList<RequestValidationResult>();
+        if(data.isEmpty())
+        {
+            RequestValidationResult result = new RequestValidationResult(false,"Request is Empty");
+            results.add(result);
+            return  results;
+        }
+        if(data.get("firstname").length() <=1)
```
Comment (josephjoys) [inline src/main/stgeorge/validator/RequestValidate.java]: Assign 1 to a final const value and re use every were else.

Comment (josephjoys) [reply src/main/stgeorge/validator/RequestValidate.java]: If data.get("firstname") is Null then we will get NullPointerException

File: src/main/stgeorge/validator/RequestValidate.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            RequestValidationResult result = new RequestValidationResult(false,"emailaddress is in valid");
+            result.setMessage("emailaddress is in valid");
+            results.add(result);
+        }
+        if(data.get("phonenumber").length() <10)
+        {
+            RequestValidationResult result = new RequestValidationResult(false,"phonenumber is invalid");
+            results.add(result);
+        }
+        if(data.get("case")=="contactus")
```
Comment (josephjoys) [inline src/main/stgeorge/validator/RequestValidate.java]: == will not work with String in Java. use .equals("contactus")

Also its good to use switch() for this?

File: src/main/stgeorge/validator/RequestValidate.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                results.add(result);
+            }
+            if(data.get("message").length()<1)
+            {
+                RequestValidationResult result = new RequestValidationResult(false,"Message is missing for qurbana nominations ");
+                results.add(result);
+            }
+
+        }
+        if(data.get("case")=="perunnal share")
```
Comment (josephjoys) [inline src/main/stgeorge/validator/RequestValidate.java]: Avoid space. space will some time urlencoded()

Comment (divpaul2016) [reply src/main/stgeorge/validator/RequestValidate.java]: Done

File: src/main/stgeorge/validator/RequestValidate.java (hunk 1 above)
Comment (divpaul2016) [reply src/main/stgeorge/validator/RequestValidate.java]: Ok

File: src/main/stgeorge/validator/RequestValidate.java (hunk 2 above)
Comment (divpaul2016) [reply src/main/stgeorge/validator/RequestValidate.java]: Done

File: src/main/stgeorge/Handler.java [hunk 4]
```diff
@@ -7,29 +7,33 @@
...
+        if (result.getrequestValid()) {
             Orchestrator orchestrator = new Orchestrator(event);
             orchestrator.orchestrate();
             switch (orchestrator.getTrigger()){
                 case EMAIL: sendEmail(orchestrator);
                 break;
             }
         }
-        return String.valueOf(event.size());
+        return result.getMessage();
```
Comment (josephjoys) [inline src/main/stgeorge/Handler.java]: We need to send this only for if validation fails ?

File: src/main/stgeorge/Handler.java [hunk 5]
```diff
@@ -7,29 +7,33 @@
 import stgeorge.mail.Emailer;
 import stgeorge.orchestrator.Orchestrator;
 import stgeorge.validator.LambdaValidator;
+import stgeorge.validator.RequestValidate;
+import stgeorge.validator.RequestValidationResult;
 
-import java.util.Map;
+import java.util.*;
```
Comment (josephjoys) [inline src/main/stgeorge/Handler.java]: Do not import *. just Import Map or any other class directly

**Pull request merged**